onesignalclient/notification.py

```diff
@@ -87,35 +97,49 @@ def contents(self):
 
     @contents.setter
     def contents(self, value):
+        self._contents = json.dumps(self._validate_content_dict(value))
+    
+    @property
+    def content_available(self):
+        return self._content_available
+
+    @content_available.setter
+    def content_available(self, value):
+        self._content_available = value
+    
+    def _validate_content_dict(self, value):
+        """
+        Validates dicts used for content properties.
+        Ex: headings, subtitle, contents.
+        """
         if isinstance(value, str):
             value = json.loads(value)
 
         if not isinstance(value, dict):
             raise TypeError('Value must be a dict.')
 
-        if not value.get(self.DEFAULT_LANGUAGE, False):
+        if len(value) > 0 and not value.get(self.DEFAULT_LANGUAGE, False):
             raise KeyError('Default language (%s) must be included.' % (
                 self.DEFAULT_LANGUAGE))
 
-        self._contents = json.dumps(value)
+        return value
 
     @property
     def headings(self):
         return json.loads(self._headings)
 
     @headings.setter
     def headings(self, value):
-        if isinstance(value, str):
-            value = json.loads(value)
-
-        if not isinstance(value, dict):
-            raise TypeError('Value must be a dict.')
-
-        if len(value) > 0 and not value.get(self.DEFAULT_LANGUAGE, False):
-            raise KeyError('Default language (%s) must be included.' % (
-                self.DEFAULT_LANGUAGE))
+        self._headings = self._validate_content_dict(value)
+    
```

Comment:
Can you remove these unwanted whitespaces?

Ref. 'W293:blank line contains whitespace'